trunk/includes/pardot-plugin-class.php

```diff
@@ -900,19 +955,24 @@ static function get_form_body( $args = array() ) {
 	 *
 	 * @since 1.4.1
 	 */
-	static function convert_embed_code_https( $embed_code ) {
-		if ( Pardot_Settings::get_setting( 'https' ) ) {
+	static function convert_embed_code_https($embed_code)
+	{
+		if (Pardot_Settings::get_setting('https')) {
 			/**
 			 * Look for URLs in the embed code
 			 */
 			$reg_exUrl = apply_filters("pardot_https_regex", "/(http|https)\:\/\/[a-zA-Z0-9\-\.]+\.[a-zA-Z0-9\-\.]+\.[a-zA-Z]{2,63}(\/\S*)?/");
-			preg_match( $reg_exUrl, $embed_code, $url );
-			/**
-			 * Replace whatever is there with the approved Pardot HTTPS URL
-			 */
-			$urlpieces  = parse_url($url[0]);
-			$httpsurl   = 'https://go.pardot.com' . $urlpieces['path'];
-			$embed_code = preg_replace( $reg_exUrl, $httpsurl, $embed_code );
+			preg_match($reg_exUrl, $embed_code, $url);
+
+			// Check if default domain is already HTTPS
+			if (strcasecmp(substr($url[0], 0, 8), "https://")) {
```

Comment:
This is the new logic that allows for HTTPS custom domains.  We will check the URL pulled from Pardot is already HTTPS.  If it is, we do nothing.  If it is using HTTP, we will replace it with the go.pardot.com domain.  While this could potentially [cause some issues in the future ](https://help.salesforce.com/s/articleView?id=000364239&type=1&language=en_US), it will at least load the form on a WordPress site with HTTPS enabled.